tests/system/feature_store/spark_hadoop_test_base.py

```diff
@@ -0,0 +1,154 @@
+    both as a source and a target. It is not designed for testing Spark operators.
+
+    To conduct a test with a specific filesystem, you should derive your test class from `SparkHadoopTestBase`.
+    In your test class, override the `custom_setup_class()` method following the example provided below.
+    Then, in your test, ensure to call the `do_test()` method of this class.
+
+    Here's an illustration of how a derived test class might look:
+    class TestFeatureStoreSparkEngine(SparkHadoopTestBase):
+        @classmethod
+        def custom_setup_class(cls):
+            cls.configure_namespace("mytest")
+            cls.configure_image_deployment(
+                Deployment.Remote,
+                "azure-storage-blob git+https://github.com/mlrun/mlrun.git@development",
+            )
+        def test_basic_remote_spark_ingest_ds_basic(self):
+            bucket = 'my_bucket'
+            ds_profile = DatastoreProfileBasic(name=self.ds_profile_name)
+            register_temporary_client_datastore_profile(ds_profile)
+            self.project.register_datastore_profile(ds_profile)
+            self.ds_upload_src(ds_profile, bucket)
+            self.do_test(
+                self.ds_src_path(ds_profile, bucket),
+                self.ds_target_path(ds_profile, bucket),
+            )
+
+    It is also possible to run the tests locally if you have pyspark installed. To run locally, call
+    configure_image_deployment(deployment_type=Deployment.Local). This can be very useful for debugging.
+    """
+
+    @classmethod
+    def configure_namespace(cls, test_suite_name="hadoop"):
+        cls.project_name = "fs-system-spark-engine-" + test_suite_name
+        cls.ds_profile_name = "ds_profile_spark_" + test_suite_name
+        cls.remote_function_name = (
+            "remote-spark-default-image-deploy-temp-" + test_suite_name
+        )
+        cls.spark_service_name = "dummy-spark-" + test_suite_name
+        cls.pq_source = "testdata.parquet"
+        cls.pq_target = "testdata_target"
+
+    @classmethod
+    def configure_image_deployment(
+        cls,
+        deployment_type=Deployment.Remote,
+        additional_pip_pacakges=None,
+        spark_image_deployed=False,
+    ):
+        cls.deployment_type = deployment_type
+        if cls.deployment_type == Deployment.Local:
+            cls.spark_image_deployed = True
+        else:
+            cls.spark_image_deployed = spark_image_deployed
+
+        if not cls.spark_image_deployed:
+            from mlrun import get_run_db
+            from mlrun.run import new_function
+
+            sj = new_function(kind="remote-spark", name=cls.remote_function_name)
+            if additional_pip_pacakges:
+                sj.spec.build.commands = [f"pip install {additional_pip_pacakges}"]
+            sj.spec.build.image = "." + cls.remote_function_name
+            sj.with_spark_service(spark_service=cls.spark_service_name)
+            sj.spec.image_pull_policy = "Always"
```

Comment:
Does this actually have any effect in this context? I think it probably doesn't...

Reply:
Somehow without this setting the test did not fetch the image

Reply:
It doesn't really make sense though... This function just deploys the image. It's not used by the test.